lib/stoplight/data_store/redis.rb

```diff
@@ -108,7 +122,9 @@ def set_last_notification(light, from_color, to_color)
       end
 
       def query_failures(light, transaction: @redis)
-        transaction.lrange(failures_key(light), 0, -1)
+        window_start = Time.now.to_i - light.window_size
+
+        transaction.zrange(failures_key(light), Float::INFINITY, window_start, rev: true, by_score: true)
```

Comment:
I'm not quite sure how it should work here. Should it me something like this?
```ruby
transaction.zrange(failures_key(light), window_start, light.window_size, rev: true, by_score: true)
```

Reply:
In the previous stoplight version, we used a List to store failures. The most recent failures were at the head of the list. 

Now we store errors in a sorted set, and we sort them by time. The most recent failures at the end of the set. 

To maintain the same interface (return recent errors at the beginning) I use `rev: true` options here. In combination with `by_score: true` it performs a reverse search by error time. According to the [documentation](https://redis.io/commands/zrange/#reverse-ranges), if you perform a reverse search you put the beginning of the range as a first argument: For instance:

```
ZRANGE zset 10 5 REV BYSCORE
```

> Will return all elements with scores less than 10 and greater than 5.